docs/intro-react.md

```diff
@@ -0,0 +1,613 @@
+---
+id: intro-react
+title: Key React Concepts
+---
+
+React Native runs on [React](https://reactjs.org/), a popular open source library for building user interfaces with JavaScript. To make the most of React Native, it helps to understand React itself. This section can get you started or can serve as a refresher course.
+
+We’re going to cover the core concepts behind React:
+
+- components
+- JSX
+- custom components
+- props
+- state
+
+If you want to dig deeper, we encourage you to check out [React's official documentation](https://reactjs.org/docs/getting-started.html).
+
+## Your first component
+
+The rest of this introduction to React uses pets in its examples: friendly, approachable creatures that need names. Your first component is a Pet:
+
+<div class="toggler">
+  <ul role="tablist" id="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTab('syntax', 'functional')">
+      Functional Syntax
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional webNote androidNote" />
+
+```SnackPlayer name=JSX
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function MyPet() {
+  return (
+    <Text>Hello, I am your pet!</Text>
+  );
+}
+```
+
+<block class="classical webNote androidNote" />
+
+```SnackPlayer name=Hello%20World
+import React, { Component } from 'react';
+import { Text } from 'react-native';
+
+export default class MyPet extends Component {
+    render() {
+      return (
+          <Text>Hello, I am your pet!</Text>
+      );
+    }
+}
+```
+
+<block class="classical functional webNote androidNote" />
+
+To define your `MyPet` component, first use JavaScript’s [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to import React and React Native’s [`Text`](/react-native/docs/next/text) component:
+
+```jsx
+import React from 'react';
+import {Text} from 'react-native';
```

Comment:
nit: other code snippets seem to include spacing around imports, eg: `import { Text }`